pilot/pkg/proxy/envoy/watcher.go

```diff
@@ -75,6 +72,28 @@ func (w *watcher) Run(ctx context.Context) {
 
 	go watchCerts(ctx, certDirs, watchFileEvents, defaultMinDelay, w.SendConfig)
 
+	shouldKickStart := true
```

Comment:
nit: maybe move this to a method? that might simplify the logic to break out of the loop ... you can just return.